src/workerd/io/worker-entrypoint.c++

```diff
@@ -395,14 +414,35 @@ kj::Promise<WorkerInterface::AlarmResult> WorkerEntrypoint::runAlarm(
       }));
     });
   });
+
+  if (isPredictableModeForTest()) {
+    promise = promise.then([worker = kj::atomicAddRef(context.getWorker())](auto res) {
+      auto lock = worker->getIsolate().getApiIsolate().lock();
+      lock->requestGcForTesting();
+      return res;
+    });
+  }
+
+  return promise;
 }
 
 kj::Promise<WorkerInterface::CustomEvent::Result>
     WorkerEntrypoint::customEvent(kj::Own<CustomEvent> event) {
   auto incomingRequest = kj::mv(KJ_REQUIRE_NONNULL(this->incomingRequest,
                                 "customEvent() can only be called once"));
   this->incomingRequest = nullptr;
-  return event->run(kj::mv(incomingRequest), entrypointName).attach(kj::mv(event));
+  
+  auto& context = incomingRequest->getContext();
+  auto promise = event->run(kj::mv(incomingRequest), entrypointName).attach(kj::mv(event));
+
+  if (isPredictableModeForTest()) {
+    promise = promise.then([worker = kj::atomicAddRef(context.getWorker())](auto res) {
```

Comment:
Since `incomingRequest` ownership was passed away to `event->run()`, `context` could be UAF after it returns.

```suggestion
  auto& context = incomingRequest->getContext();
  auto worker = kj::atomicAddRef(context.getWorker());
  auto promise = event->run(kj::mv(incomingRequest), entrypointName).attach(kj::mv(event));

  if (isPredictableModeForTest()) {
    promise = promise.then([worker = kj::mv(worker)](auto res) {
```

Reply:
:+1: